src/core/FileWatcher.cpp

```diff
@@ -121,32 +128,33 @@ void FileWatcher::checkFileChanged()
     m_ignoreFileChange = true;
 
     // Only trigger the change notice if there is a checksum mismatch
-    auto checksum = calculateChecksum();
+    m_mutex.lock();
```

Comment:
Why not use a QMutexLocker here?

Reply:
Since there is only on code path I decided not to allocate a new variable for this purpose.

Reply:
That may not be so in the future. Better be safe than sorry.